lib/dataSource.js

```diff
@@ -16,138 +16,123 @@ var fluid = require("infusion"),
     kettle = fluid.registerNamespace("kettle"),
     fs = require("fs"),
     http = require("http"),
-    url = require("url");
+    urlModule = require("url"),
```

Comment:
We can't currently reuse any of the code found here in the [browser-side `dataSource` grade](https://issues.gpii.net/browse/GPII-1587) we have discussed previously.  For everything that uses a `require` here, we need a "browser safe" and "node specific" version which are located in separate files.  In some cases it's possible to have the "node specific" version extend the "browser safe" version, in other cases, you need a third "common" grade that both can safely extend.  AJV is a natural alternative to JSONlint, as we have already demonstrated its use cross-environment.  This is a longer-term effort, but we can lay the groundwork by at least leaving `TODO` comments regarding cross-environment options.